### PR TITLE
Check issue template

### PR DESCRIPTION
**Pull request by codebytere**
Todo:
- [x] add to ReadMe

/cc @hiimbex

### CODE REVIEW

File: lib/checkTemplateMatch.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const res = await context.github.repos.getContent(this.repo({
+  path: '.github/ISSUE_TEMPLATE.md'
+}))
+
+const issueTemplate = Buffer.from(res.data.content, 'base64').toString()
+
+function checkTemplateMatch(body) {
+  const lines = issueTemplate.split("\n*")
+  lines.forEach(line => {
+    if (!body.includes(line)) {
```
Comment (j-f1) [inline lib/checkTemplateMatch.js]: How does this work for templates with forms you have to fill out? For example:

```markdown
**Foo version**: [enter your version of Foo here]
**OS version**: [enter your operating system’s name and version here]
```

File: lib/getComment.js [hunk 2]
```diff
@@ -1,17 +1,13 @@
-module.exports = getComment
-
 function getComment (comment, defaultComment) {
-  // comment can be either a string or an array
-
-  if (!comment) {
-    comment = defaultComment
-  }
+  // set comment to default if none passed it
+  const eventComment = (!comment) ? defaultComment : comment
```
Comment (hiimbex) [inline lib/getComment.js]: for future reference, I'd prefer style changes like these (basically all of `lib/getComment.js`) in a separate PR.

File: test/events/issueTemplateFailEvent.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        "name": "testing-things",
+        "owner": {
+          "login": "hiimbex"
+        }
+      },
+      "installation": {
+        "id": 35471
+      }
+    }
+}
```
Comment (hiimbex) [inline test/events/issueTemplateFailEvent.json]: this fixture is identical to `test/events/issueSuccessEvent.json`, but I think ideally eventually the test will look more like an offshoot of a specific issue template

File: index.js [hunk 4]
```diff
@@ -30,13 +32,19 @@ module.exports = robot => {
         }
       }
 
+      if (config.matchIssueTemplate) {
```
Comment (hiimbex) [inline index.js]: 👍 to this config name, make sure to add it to the readme

File: lib/checkTemplateMatch.js (hunk 1 above)
Comment (hiimbex) [reply lib/checkTemplateMatch.js]: Right now this only checks that you correctly include all of the lines, as if you were filling out a form, but I'm not sure that's the best approach. Some times issue templates follow a branching style, like if it's a bug fill out area A, a feature request area B, etc.

As I said before, I think a solid approach for step one would be checking that the issue body isn't exactly the issue template ( ie `!body.includes(issueTemplate)`)

File: lib/checkTemplateMatch.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+const res = await context.github.repos.getContent(this.repo({
+  path: '.github/ISSUE_TEMPLATE.md'
+}))
```
Comment (hiimbex) [inline lib/checkTemplateMatch.js]: also I think CI is failing due to this needing to be inside something async

File: test/events/issueTemplateFailEvent.json (hunk 3 above)
Comment (codebytere) [reply test/events/issueTemplateFailEvent.json]: yeah oops i know sorry i have yet to change it for issue template purposes

File: lib/checkTemplateMatch.js (hunk 1 above)
Comment (codebytere) [reply lib/checkTemplateMatch.js]: so this is what i thought about earlier, and `!body.includes(issueTemplate)` fails, if you look at this example:
```js
const body = ' Operating System Mac Version 1.7.6 Bug Description Wow  i have a bug!!!'
const issueTemplate = ' Operating System Version Bug Description'
```
`!body.includes(issueTemplate)` would return `false` here

File: lib/getComment.js (hunk 2 above)
Comment (codebytere) [reply lib/getComment.js]: oh sorry, would you prefer i pull it out and make it a separate PR?

File: README.md [hunk 6]
```diff
@@ -37,6 +37,9 @@ requestInfoDefaultTitles:
 # *OPTIONAL* Label to be added to Issues and Pull Requests with insufficient information given
 requestInfoLabelToAdd: needs-more-info
 
+# *OPTIONAL* Require the issue to match an issue template you have in your repository.
```
Comment (hiimbex) [inline README.md]: Do you think maybe this could be better rephrased to explain what it's actually checking?

**Comment by hiimbex** [comment]
One last nit comment and then this should be good to go!

**Comment by hiimbex** [comment]
Actually, any chance you wanna add a test for this?

**Comment by hiimbex** [comment]
hey @codebytere no rush, but any interest in finishing this up?

**Comment by codebytere** [comment]
@hiimbex would https://github.com/behaviorbot/request-info/pull/32/files not take its place?

**Comment by shiftkey** [comment]
I think this should be covered by #39 which was just merged. @codebytere let me know if there's anything more do to with that.

**Pull request closed without merging**